pkg/util/jsonpath/parser.go

```diff
@@ -382,19 +382,21 @@ Loop:
 // parseField scans a field until a terminator
 func (p *Parser) parseField(cur *ListNode) error {
 	p.consumeText()
-	var r rune
+Loop:
 	for {
-		r = p.next()
-		if isTerminator(r) {
+		switch r := p.next(); {
```

Comment:
rather than do the labelled breaking, I'd introduce a new helper function:

```go
func advance(p *Parser) bool {
  switch r := p.next(); {
     ...
  }
}

...
for advance(p) {}
...
```